src/peergos/server/LinkIdentity.java

```diff
@@ -0,0 +1,83 @@
+package peergos.server;
+
+import peergos.server.util.*;
+import peergos.shared.*;
+import peergos.shared.crypto.asymmetric.*;
+import peergos.shared.crypto.hash.*;
+import peergos.shared.crypto.symmetric.*;
+import peergos.shared.io.ipfs.multibase.*;
+import peergos.shared.user.*;
+import peergos.shared.user.fs.*;
+
+import java.io.*;
+import java.nio.file.*;
+import java.util.*;
+
+public class LinkIdentity {
+
+    /** Link a peergos account to an account on an external service,
+     *  where you can post textual content up to 280 characters long.
+     *
+     * @param a
+     * @param network
+     * @param crypto
+     */
+    public static void link(Args a, NetworkAccess network, Crypto crypto) {
+        String username = a.getArg("username");
+        Console console = System.console();
+        String password = new String(console.readPassword("Enter password for " + username + ":"));
+        UserContext context = UserContext.signIn(username, password, network, crypto).join();
+        String usernameB = a.getArg("service-username");
+        String serviceB = a.getArg("service");
+        boolean encrypted = a.getBoolean("encrypted");
+        boolean publish = ! encrypted && a.getBoolean("publish", false);
```

Comment:
Is that logic correct?

Reply:
An encrypted one shouldn't be published as that would defeat the point of encrypting it.